v2/internal/controllers/crd_dataprotection_backuppolicy_test.go

```diff
@@ -0,0 +1,218 @@
+	tc.Expect(backuppolicy.Status.Properties.BackupPolicy.PolicyRules[0].AzureBackup.Name).To(BeEquivalentTo(&AzureBackRule_Name))
+	tc.Expect(backuppolicy.Status.Properties.BackupPolicy.PolicyRules[0].AzureBackup.ObjectType).To(BeEquivalentTo(&AzureBackupRule_ObjectType))
+
+	// Assertions and Expectations for BackupPolicy:AzureBackupRule:BackupParameters
+	tc.Expect(backuppolicy.Status.Properties.BackupPolicy.PolicyRules[0].AzureBackup.BackupParameters.AzureBackupParams.BackupType).To(BeEquivalentTo(&AzureBackupParams_BackupType_Value))
+	tc.Expect(backuppolicy.Status.Properties.BackupPolicy.PolicyRules[0].AzureBackup.BackupParameters.AzureBackupParams.ObjectType).To(BeEquivalentTo(&AzureBackupParams_ObjectType_Value))
+
+	// Assertions and Expectations for BackupPolicy:AzureBackupRule:DataStore
+	tc.Expect(backuppolicy.Status.Properties.BackupPolicy.PolicyRules[0].AzureBackup.DataStore.DataStoreType).To(BeEquivalentTo(&DataStore_DataStoreType_Value))
+	tc.Expect(backuppolicy.Status.Properties.BackupPolicy.PolicyRules[0].AzureBackup.DataStore.ObjectType).To(BeEquivalentTo(&DataStore_ObjectType_Value))
+
+	// Assertions and Expectations for BackupPolicy:AzureBackupRule:Trigger
+	tc.Expect(backuppolicy.Status.Properties.BackupPolicy.PolicyRules[0].AzureBackup.Trigger.Schedule.ObjectType).To(BeEquivalentTo(&Schedule_ObjectType_Value))
+
+	// Assertions and Expectations for BackupPolicy:AzureRetentionRule
+	tc.Expect(backuppolicy.Status.Properties.BackupPolicy.PolicyRules[1].AzureRetention.Name).To(BeEquivalentTo(&AzureRetentionRule_Name))
+	tc.Expect(backuppolicy.Status.Properties.BackupPolicy.PolicyRules[1].AzureRetention.ObjectType).To(BeEquivalentTo(&AzureRetentionRule_ObjectType))
+	tc.Expect(backuppolicy.Status.Properties.BackupPolicy.PolicyRules[1].AzureRetention.IsDefault).To(BeEquivalentTo(&AzureRetentionRule_IsDefault))
+
+	// Assertions and Expectations for BackupPolicy:AzureRetentionRule:Lifecycles
+	tc.Expect(backuppolicy.Status.Properties.BackupPolicy.PolicyRules[1].AzureRetention.Lifecycles[0].DeleteAfter.AbsoluteDeleteOption.Duration).To(BeEquivalentTo(&AzureRetentionRule_Lifecycles_DeleteAfter_Duration))
+	tc.Expect(backuppolicy.Status.Properties.BackupPolicy.PolicyRules[1].AzureRetention.Lifecycles[0].DeleteAfter.AbsoluteDeleteOption.ObjectType).To(BeEquivalentTo(&AzureRetentionRule_Lifecycles_DeleteAfter_ObjectType))
+
+	// Assertions and Expectations for BackupPolicy:AzureRetentionRule:Lifecycles:SourceDataStore
+	tc.Expect(backuppolicy.Status.Properties.BackupPolicy.PolicyRules[1].AzureRetention.Lifecycles[0].SourceDataStore.DataStoreType).To(BeEquivalentTo(&AzureRetentionRule_Lifecycles_SourceDataStore_DataStoreType))
+	tc.Expect(backuppolicy.Status.Properties.BackupPolicy.PolicyRules[1].AzureRetention.Lifecycles[0].SourceDataStore.ObjectType).To(BeEquivalentTo(&AzureRetentionRule_Lifecycles_SourceDataStore_ObjectType))
+
+	tc.Expect(backuppolicy.Status.Id).ToNot(BeNil())
+
```

Comment:
Also, as the name mentions `CRUD` test - would be good to add a simple PATCH as well in the test for all the resources. You can refer to the other tests for this.

Reply:
Glad you noticed, added a simple PATCH Operation on the Backup vault test file.
However, the Backup Policy doesn't allow any updates. Thus, no such PATCH operation is needed there.

Reply:
ASO PATCH != RP PATCH.

Does backupPolicy support subsequent PUT updates at the RP level? I would imagine yes, to allow users to change things like retention?

If so, you should be able to do a PATCH (to the Kubernetes resource) which results in a fresh PUT to the RP which updates the backupPolicy retention.

Reply:
Afaik, Backup Policy does not support any update, including the retention period. This has been confirmed by trying over Az Portal as well as discussing internally. Please let me know if we're missing anything that you wanted to say.

Reply:
Ahh, this makes sense. We have #1443 tracking write-once properties that can't be changed after the resource is created. You can skip the Patch part of the test.